cpp/ql/src/semmle/code/cpp/models/implementations/StdContainer.qll

```diff
@@ -0,0 +1,102 @@
+/**
+ * Provides models for C++ containers `std::array`, `std::vector`, `std::deque`, `std::list` and `std::forward_list`.
+ */
+
+import semmle.code.cpp.models.interfaces.Taint
+
+/**
+ * Additional model for standard container constructors that reference the
+ * value type of the container (that is, the `T` in `std::vector<T>`).  For
+ * example the fill constructor:
+ * ```
+ * std::vector<std::string> v(100, potentially_tainted_string);
+ * ```
+ */
+class StdContainerConstructor extends Constructor, TaintFunction {
+  StdContainerConstructor() {
```

Comment:
How about we rename this to `StdSequenceContainerConstructor` to follow the naming scheme from https://en.cppreference.com/w/cpp/container? Then we can name the unordered containers `AssociativeContainer` and `UnorderedAssociativeContainers` once we add those in the future.

I guess this comment applies to many places in this PR.

Reply:
I think that's a more descriptive name, I'll make the change...

Reply:
Done.